### PR TITLE
Visualize routes with multiple services on overview

### PR DESCRIPTION
**Pull request by spadgett**
https://trello.com/c/Eddu8kg4

openshift_web_console

/cc @jwforres

### CODE REVIEW

**Comment by spadgett** [comment]
@jwforres without the gaps

screen shot 2016-07-07 at 3 20 12 pm

**Comment by liggitt** [comment]
do we want something that will work with more than two services?

**Comment by spadgett** [comment]
@liggitt Yes, we fall back to a different view if there are more than two. @ajacobs21e is working on designs, but currently we just show the progress bar on the right for each.

File: app/views/overview/_dc.html [hunk 1]
```diff
@@ -4,9 +4,13 @@
...
       <a ng-href="{{service | navigateResourceURL}}">{{service.metadata.name}}</a>
     </div>
-    <div>
+    <div class="service-metadata">
       <span ng-if="!scalableDeploymentByConfig[dcName]">Deploying...</span>
-      <relative-timestamp ng-if="scalableDeploymentByConfig[dcName]" timestamp="scalableDeploymentByConfig[dcName].metadata.creationTimestamp"></relative-timestamp>
+      <div ng-if="scalableDeploymentByConfig[dcName] && weightByService[service.metadata.name]">
+        <ng-include src="'views/overview/_traffic-percent.html'"></ng-include>
+      </div>
+      <!-- TODO: Move under the deployment tile -->
```
Comment (jwforres) [inline app/views/overview/_dc.html]: assume we need to do this TODO still

Comment (spadgett) [inline app/views/overview/_dc.html]: Yeah, having a hard time finding a place for it though.

File: app/views/overview/_traffic-percent.html [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<div ng-if="alternateServices.length">
+  <span class="visible-xs visible-sm">
+    Traffic {{(weightByService[service.metadata.name] / totalWeight) | percent}}
```
Comment (jwforres) [inline app/views/overview/_traffic-percent.html]: probably a chance of us having a divide by zero here once we get them to allow zero for weight, unless we require at least one non-zero weight

File: app/views/overview/_traffic-percent.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  </span>
+  <div class="hidden-xs hidden-sm">
+    <span class="traffic-label">Traffic</span>
+    <!-- Calculate the width of the progress bar in px (max 250px) -->
+    <div class="progress progress-sm"
+         ng-style="{ width: ((weightByService[service.metadata.name] / totalWeight * 250) | number) + 'px'}">
```
Comment (jwforres) [inline app/views/overview/_traffic-percent.html]: same here, divide by zero chance

**Comment by spadgett** [comment]
Suggestion from @ajacobs21e to use different shades of blue for each service when more than 2. @jwforres @ajacobs21e Thoughts? I think it works when there are 2 services and the bars touch, but not sure about here. We also use different blues to indicate pod status (not ready, running, terminating). I wonder if people will think it's the same for services.

screen shot 2016-07-11 at 1 01 21 pm

**Comment by spadgett** [comment]
Same blue for comparison:

screen shot 2016-07-11 at 1 08 16 pm

**Comment by jwforres** [comment]
hmm the different blues when they are separated look weird to me... looking at it I would expect that the shade means something, possibly that it coincides with the percentage of traffic that service is receiving, like all of those ones that were 17% would be the same shade

**Comment by spadgett** [comment]
@jwforres OK, this is probably ready for another look.

**Comment by jwforres** [comment]
[merge]

**Comment by ajacobs21e** [comment]
> We also use different blues to indicate pod status (not ready, running, terminating)

@spadgett Do you have a master list of all the possible states and how they're visualized in the UI (color, icon, etc)?

**Comment by spadgett** [comment]
@ajacobs21e here are the statuses and colors for the donut

https://github.com/openshift/origin-web-console/blob/1971a40cd061f1095e54043f0d5a859f80448476/app/scripts/directives/podDonut.js#L91-L104

**Pull request merged**